networks/p2p/dial.go

```diff
@@ -490,6 +491,10 @@ func (t *dialTask) Do(srv Server) {
 		err = t.dialMulti(srv, t.dest)
 	} else {
 		err = t.dial(srv, t.dest)
+		// redial with updated connection
+		if err == errUpdateDial {
+			err = t.dialMulti(srv, t.dest)
+		}
```

Comment:
We need to add this code in the following retry logic.

Reply:
You mean deleting `srv.AddPeer(dialDest)`?

Reply:
I mean the `t.dial()` in the following code. 
https://github.com/klaytn/klaytn/blob/7f51206cc5a5cca75405175847dc4b731beb7684/networks/p2p/dial.go#L500-L509

Reply:
The below catchs failing cases. It could catch the case when the updated dial fails too.
Do you still think it should be inside the `if` statement below?

Reply:
`errUpdateDial` is not `dialError`. Without the error handling logic, a connection cannot update to a multi-channel connection if it failed to dial first and then succeed to dial a second time.

Reply:
What do you think of :
```
for err != nil {
		switch err.(type) {
		case *dialError:
			logger.Debug("[Dial] Failed dialing", "task", t, "err", err)
			// Try resolving the ID of static nodes if dialing failed.
			if t.flags&staticDialedConn != 0 && t.resolve(srv, t.dest.NType) {
				if t.dest.TCPs != nil && len(t.dest.TCPs) != 0 {
					err = t.dialMulti(srv, t.dest)
				} else {
					err = t.dial(srv, t.dest)
				}
			}
			if err != nil {
				t.failedTry++
			}
			return
		case errUpdateDial:
			err = t.dialMulti(srv, t.dest)
		case error:
			logger.Debug("[Dial] Failed setupConn", "task", t, "err", err)
			t.failedTry++
			return
		}
	}
```

Reply:
No.
```
	if err != nil { // this line may be removable
		logger.Debug("[Dial] Failed dialing", "task", t, "err", err)
		// Try resolving the ID of static nodes if dialing failed.
		if _, ok := err.(*dialError); ok && t.flags&staticDialedConn != 0 {
			if t.resolve(srv, t.dest.NType) {
				if t.dest.TCPs != nil && len(t.dest.TCPs) != 0 {
					err = t.dialMulti(srv, t.dest)
				} else {
					err = t.dial(srv, t.dest)
					// redial with updated connection
					if err == errUpdateDial {
						_ = t.dialMulti(srv, t.dest)
					}
				}
				if err != nil {
					t.failedTry++
				}
			} else {
				t.failedTry++
			}
		} else {
			t.failedTry++
		}
	}
```

Reply:
@winnie-byun I can't follow your context. Could you make a short meeting together?

Reply:
@aidan-kwon @KimKyungup 
What do you think of the following code for more clean syntax? 
```
	var err error
	if t.dest.TCPs != nil && len(t.dest.TCPs) != 0 {
		err = t.dialMulti(srv, t.dest)
	} else {
		err = t.dial(srv, t.dest)
	}

	if err != nil {
		logger.Debug("[Dial] Failed dialing", "task", t, "err", err)
		// Try resolving the ID of static nodes if dialing failed.
		if _, ok := err.(*dialError); ok && t.flags&staticDialedConn != 0 && t.resolve(srv, t.dest.NType) {
			if t.dest.TCPs != nil && len(t.dest.TCPs) != 0 {
				err = t.dialMulti(srv, t.dest)
			} else {
				err = t.dial(srv, t.dest)
			}
			if err != nil {
				t.failedTry++
			}
		} else {
			t.failedTry++
		}

		// redial with updated connection
		if err == errUpdateDial {
			err = t.dialMulti(srv, t.dest)
		} else {
			t.failedTry++
		}
	}
```

Reply:
I like this. :)

Reply:
@winnie-byun Semantic is a little bit changed, but it's good to go. Could you refine the definition of `failedTry`? This logic can increase `failedTry` twice.

Reply:
@aidan-kwon You are right. It should increase only once. I changed a little bit to do it once.
PTAL :)